src/Driver/Http2Driver.php

```diff
@@ -706,27 +707,37 @@ private function parser(string $settings = null): \Generator
             }
 
             while (true) {
-                if (++$frameCount === $maxFramesPerSecond) {
-                    $now = $this->timeReference->getCurrentTime();
-                    if ($lastReset === $now && $bytesReceived / $frameCount < $minAverageFrameSize) {
+                $now = $this->timeReference->getCurrentTime();
+                if ($lastReset === $now) {
+                    // Inspired by nginx flood detection:
+                    // https://github.com/nginx/nginx/commit/af0e284b967d0ecff1abcdce6558ed4635e3e757
+                    if ($totalBytesReceivedSinceReset > $payloadBytesReceivedSinceReset + 8192) {
```

Comment:
A large header might exceed this, no?

Reply:
Yes, headers and continuation frames are considered payload data.